examples/complex/xss_scanner.py

```diff
@@ -380,33 +383,38 @@ def handle_starttag(self, tag, attrs):
                            injection_point,
                            " onmouseover=alert(0) t=",
                            match.decode('utf-8'))
-        elif in_HTML and not in_script and inject_open_angle and inject_close_angle and inject_slash:  # e.g. <html>PAYLOAD</html>
+        elif in_HTML_val and not in_script_val and inject_open_angle and inject_close_angle and inject_slash:  # e.g. <html>PAYLOAD</html>
             return XSSData(request_URL,
                            injection_point,
                            '<script>alert(0)</script>',
                            match.decode('utf-8'))
         else:
             return None
+    return None
 
 
 # response is mitmproxy's entry point
 def response(flow: http.HTTPFlow) -> None:
     cookiesDict = get_cookies(flow)
+    try:
+        resp = flow.response.content.decode('utf-8')
```

Comment:
Again, you assume that the response content is utf8, which is not guaranteed to be true. In fact It's latin-1 by default. Use `.get_text(strict=False)` here.